### PR TITLE
Exit with usage code when incorrect args are used

### PR DESCRIPTION
**Pull request by corwinsheahan-wf**
## Motivation
  
When non-recognized args are used, we print the help message, but still exit with a `0` exit code. This can cause false positives in CI if legacy command line args are used.
## Changes
  
* Return appropriate exit code
* Add test
## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/dependency_validator/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/dependency_validator/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by evanweible-wf — APPROVED** [review]

**Comment by corwinsheahan-wf** [comment]
@evanweible-wf Updated.

**Review by evanweible-wf — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# 3.2.0
+
+- Return non-zero exit code from executable when incorrect args are used
```
Comment (evanweible-wf) [inline CHANGELOG.md:3]: Sorry, this is a [nit] but this should be a patch release since we're treating it as a bug fix

Comment (corwinsheahan-wf) [reply CHANGELOG.md:3]: Updated

**Review by evanweible-wf — APPROVED** [review]

**Review by evanweible-wf — APPROVED** [review]

**Comment by evanweible-wf** [comment]
QA +1

@Workiva/release-management-pp

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**